lib/logstash/inputs/tzinfo_jruby_patch.rb

```diff
@@ -0,0 +1,39 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+require 'tzinfo'
+
+if defined?(TZInfo::VERSION) && TZInfo::VERSION > '2'
+  module TZInfo
+    # A time represented as an `Integer` number of seconds since 1970-01-01
+    # 00:00:00 UTC (ignoring leap seconds), the fraction through the second
+    # (sub_second as a `Rational`) and an optional UTC offset. Like Ruby's `Time`
+    # class, {Timestamp} can distinguish between a local time with a zero offset
+    # and a time specified explicitly as UTC.
+    class Timestamp
+
+      protected
+
+      def new_datetime(klass = DateTime)
```

Comment:
Is this a monkey patch of an existing method or is this a new method added? This should be documented. If this is a new method, is there a specific reason to add it in TZInfo::Timestamp?

Reply:
It is a patch on the TZInfo::Timestamp class, I'm redefining the `new_datetime` method.